src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/SpecimenCollectionDateTime.java

```diff
@@ -0,0 +1,100 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+/**
+ * Example 1: DTM+SCO:20100223:102'
+ * Example 2: DTM+SCO:201002231541:203'
+ */
+@Getter
+@EqualsAndHashCode(callSuper = false)
+@Builder
+public class SpecimenCollectionDateTime extends Segment {
+    protected static final String KEY = "DTM";
+    private static final String QUALIFIER = "SCO";
+    private static final String KEY_QUALIFIER = KEY + PLUS_SEPARATOR + QUALIFIER;
+    private static final DateTimeFormatter DATE_FORMATTER_CCYYMMDD = DateTimeFormatter.ofPattern("yyyyMMdd");
+    private static final DateTimeFormatter DATE_FORMATTER_CCYYMMDDHHMM = DateTimeFormatter.ofPattern("yyyyMMddHHmm");
+
+    @NonNull
+    private final String collectionDateTime;
+    @NonNull
+    private final DateFormat dateFormat;
+
+    public static SpecimenCollectionDateTime fromString(final String edifactString) {
+        if (!edifactString.startsWith(SpecimenCollectionDateTime.KEY_QUALIFIER)) {
+            throw new IllegalArgumentException(
+                "Can't create " + SpecimenCollectionDateTime.class.getSimpleName() + " from " + edifactString);
+        }
+        final String input = Split.byPlus(edifactString)[1];
+        final String collectionDateTime = Split.byColon(input)[1];
+        final String format = Split.bySegmentTerminator(Split.byColon(input)[2])[0];
+        final SpecimenCollectionDateTimeBuilder collectionDateTimeBuilder = SpecimenCollectionDateTime.builder();
+
+        if (isNotBlank(collectionDateTime) && isNotBlank(format)) {
```

Comment:
Is it better now? 766d7ee20a5da7c0a799acd52e137669409ac8a3
Thanks!

Reply:
This still allows for empty segments like `DTM+SCO::`. I would suggest

```
if (isBlank(collectionDateTime) || isBlank(format)) {
    throw new IllegalArgumentException("Can't create SpecimenCollectionDateTime from " + edifactString
        + " because of missing date-time and/or format definition");
}
final String formattedFhirDate = getFormattedFhirDate(collectionDateTime, format);
collectionDateTimeBuilder
    .collectionDateTime(formattedFhirDate)
    .dateFormat(DateFormat.fromCode(format));
return collectionDateTimeBuilder.build();

Reply:
Applied your suggestions in 34d46b1132730fd013f30b8bbe47809a5f0e584d
Thanks!